### PR TITLE
build: add bazel build

### PR DESCRIPTION
**Pull request by jbedard**
The use of the async sass API currently seems to break when dealing with the bazel/rules_js symlinking of npm modules so the `NG_BUILD_LEGACY_SASS` flag has been set to use the old API. This was a recent change switching to the async API: https://github.com/angular/angular-cli/commit/308e3a017f876bfc727e68803bfbce11e9d3396e

Depends on:
* ~~https://github.com/angular/material.angular.io/pull/1173~~
* https://github.com/angular/material.angular.io/pull/1172

### CODE REVIEW

File: .circleci/config.yml
Comment (josephperrott) [inline .circleci/config.yml:57]: Are we planning to do this still?

File: .npmrc
Comment (josephperrott) [inline .npmrc:1]: Why is this necessary, should we be using resolutions instead?

File: scenes/e2e/protractor.conf.js
Comment (josephperrott) [inline scenes/e2e/protractor.conf.js:21]: const?

File: .npmrc
Comment (jbedard) [reply .npmrc:1]: I believe yarn defaults to false while pnpm defaults to true so it needs to be set to use rules_js.

If using resolutions isn't required then I'm not sure if it should be done in this PR?

File: .circleci/config.yml
Comment (jbedard) [reply .circleci/config.yml:57]: After talking offline... it's not done today so we're not doing it in bazel.

File: scenes/src/app/scenes/slider/slider-scene.ts
Comment (josephperrott) [inline scenes/src/app/scenes/slider/slider-scene.ts:16]: Is this as any necessary?  `MatSlider` should have a `focus` method.

**Review by josephperrott — APPROVED** [review]

File: package.json
Comment (devversion) [inline package.json:101]: Do you know where nrwl/devkit is coming from? That looks unexpected

File: scenes/karma.conf.js
Comment (devversion) [inline scenes/karma.conf.js:47]: I don't think we use Travis. Is this an artifact from some testing?

**Review by devversion — APPROVED** [review]
I also went through it. Looks great overall, just some minor nits/questions

File: scenes/karma.conf.js
Comment (jbedard) [reply scenes/karma.conf.js:47]: No I'm not sure, it already existed in all the other karma configs

File: .circleci/config.yml
Comment (jbedard) [reply .circleci/config.yml:57]: I think this was only required for e2e browser testing which is not on CI today so we're not doing it in bazel either.

File: .bazelrc
Comment (jbedard) [reply .bazelrc:33]: I assume it was only added for e2e tests so I'll try just deleting it.

File: .circleci/config.yml
Comment (devversion) [reply .circleci/config.yml]: but the plan is to enable remote execution for build and testing right? I might be misunderstanding

File: package.json
Comment (jbedard) [reply package.json]: Looks like it is no longer being used, see yarn.lock in https://github.com/angular/material.angular.io/commit/ac0acaebc46e3c29e09fa36af3d323ed049574b2#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deL3703

File: scenes/src/app/scenes/slider/slider-scene.ts
Comment (jbedard) [reply scenes/src/app/scenes/slider/slider-scene.ts:16]: Yeah I don't understand this. I don't actually see any `focus` method on the class or in the compiled .d.ts?

https://github.com/angular/components/blob/6011b7d7e789fd9ad0c724df20501c2758f750e2/src/material/slider/slider.ts#L95

**Comment by gregmagolan** [comment]
@devversion With Bazel 6.0.0rc1 remote execution now works with rules_js. It could be done as a follow-up to this PR pretty easily.

File: WORKSPACE.bazel
Comment (gregmagolan) [inline WORKSPACE.bazel:17]: Some nice per improvements have landed in rules_js since this release. Best to update to the latest of rules_js and  aspect_bazel_lib before landing.

File: scenes/karma.conf.js
Comment (devversion) [reply scenes/karma.conf.js:47]: Interesting, seems like a legacy artifact. We can clean-up ourselves, but this config did not have this travis check before, so we shouldn't add it here?

File: scenes/src/app/scenes/slider/slider-scene.ts
Comment (devversion) [reply scenes/src/app/scenes/slider/slider-scene.ts:16]: I think it might be because of recent breaking changes to the slider for Angular v15. Basically a whole rework. Maybe add a TODO here and keep the cast for now?

**Review by devversion — APPROVED** [review]
LGTM

**Comment by devversion** [comment]
@jbedard do you want to merge this into a separate branch? I think we cannot merge this into `main` for now till at least Angular v15 is out (which is expected this very soon)

File: scenes/src/app/scenes/slider/slider-scene.ts
Comment (jbedard) [reply scenes/src/app/scenes/slider/slider-scene.ts:17]: I've added the TODO

File: scenes/karma.conf.js
Comment (jbedard) [reply scenes/karma.conf.js:47]: I think I was aligning this with karma.conf.js (in the root), I don't recall why 😬

**Comment by jbedard** [comment]
>@jbedard do you want to merge this into a separate branch? I think we cannot merge this into main for now till at least Angular v15 is out (which is expected this very soon)

We can merge to a separate branch if you'd like to merge it into `main` later. It won't have any CI keeping it green in the meantime? (not that it's green atm anyway 😬).

**Comment by jbedard** [comment]
Note the first commit is also on its own in https://github.com/angular/material.angular.io/pull/1172 if you'd prefer merging that on its own. It is also broken for I think legit reasons unrelated to my change? See that PR...

**Comment by devversion** [comment]
@jbedard Yeah, separate branch might be best for now to not risk the v15 docs. We can merge to main soon then:

* You can still see CI by creating a PR from the feature branch to `main`.
* Your commit `fix TS compiler error` seems to bring in accidentally some symlinks e.g. for `blaze-bin`?
* Regarding the failing audit job:  Seems green on `main` unless I miss something & we run it periodically as a cronjob from the `angular/components` repo IIRC.

**Comment by jbedard** [comment]
I'm looking at the audit stuff more atm and will post again when I have an update, and will sort out any issues with the bazel files (sorry I'm randomly rebasing/pushing trying to get it green atm).

Shall I change this PR to create a new branch? I'm not sure the best way to do that...

**Comment by devversion** [comment]
I've created a branch for the Bazel migration. Just let me know when I should merge this PR

**Comment by jbedard** [comment]
Actually I may have fixed the audits after all this time. It seems as if the default `light-server` host is `localhost` on the cli and `0.0.0.0` with the js api, and I guess using an IP for the host really kills the: "Best Practices", "Performance", and "Progressive Web App" scores 🤔 

It's passing in https://github.com/angular/material.angular.io/pull/1172 if you want to merge it into `main` right now. I'm looking to see why it's still red under bazel here...

**Comment by jbedard** [comment]
Ok it's green under bazel as well now. Sorry for the confusion with me thinking it was a legit failure, I guess `0.0.0.0` is bad ux 😬 

You can do the merge into the branch 👍

**Pull request merged**
